west.yml

```diff
@@ -2,14 +2,25 @@ manifest:
   remotes:
     - name: zephyrproject
       url-base: https://github.com/zephyrproject-rtos
+  defaults:
+    remote: zephyrproject
 
   projects:
     # Zephyr repository (specific tag or branch)
     - name: zephyr
       path: zephyr
-      remote: zephyrproject
       revision: v3.6.0
       import: true  # Pull in Zephyr modules automatically
+    - name: cmsis
```

Comment:
Besides that, the current implementation of the west.yml will still clone all other dependencies that are not intended to be used. Go through the available properties that are used for _import_ to get only those 2 modules.

Explicit definition of the module will override the parent's definition (path/revision etc.). The other dependencies will still be cloned.

Reply:
The reset of the code LGTM!